hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/TransactionModuleTest.java

```diff
@@ -132,6 +143,19 @@ void providesNullEthTxDataIfNotEthereumTransaction() {
                 TransactionModule.maybeProvideHydratedEthTxData(context, hydration, DEFAULT_HEDERA_CONFIG, fileStore));
     }
 
+    @Test
+    void providesSystemGasContractCalculator() {
+        given(context.dispatchComputeFees(TransactionBody.DEFAULT)).willReturn(new Fees(1, 2, 3));
+        given(tinybarValues.childTransactionServiceGasPrice()).willReturn(2L);
+        given(canonicalDispatchPrices.canonicalPriceInTinycents(DispatchType.APPROVE))
+                .willReturn(66L);
+        given(tinybarValues.asTinybars(66L)).willReturn(7L);
+        final var calculator =
+                TransactionModule.provideSystemContractGasCalculator(context, canonicalDispatchPrices, tinybarValues);
+        final var result = calculator.gasRequirement(TransactionBody.DEFAULT, DispatchType.APPROVE);
+        assertEquals(4L, result);
```

Comment:
I guess the idea here is that whoever is supposed to look at this test when it fails is one who is deeply steeped in fees.  I can make a guess that we've got a `7 - (1+2)` or maybe a `7 - 3` but can't choose between them and it may be something else too.  A comment, maybe?  Or at a bare minimum choose values for the fees (`(1,2,3)`) such that there's no ambiguity about what the calculation is supposed to be, once you reverse-engineer it.

Reply:
Added comments clarifying intent.